### PR TITLE
[fx] fix the false interpretation of algorithm 3 in https://arxiv.org/abs/1604.06174.

### PR DESCRIPTION
**Pull request by super-dainiu**
### What's new?
After a discussion with my collaborator @Cypher30, I figured out my misinterpretation of algorithm 3 in https://arxiv.org/abs/1604.06174. This PR can be viewed as an amendment to this previous PR #1433. Now users don't need to give a specific budget, the budget is automatically estimated through grid search.
``` python
model = resnet18()
input_sample = torch.rand(4, 3, 224, 224)
gm = symbolic_trace(model)
MetaInfoProp(gm).run(input_sample)
gm = chen_greedy(gm)
```

### CODE REVIEW

File: colossalai/fx/passes/algorithms/ckpt_solver_chen.py [hunk 1]
```diff
@@ -1,45 +1,71 @@
...
+        Search ckpt strategy with B = 0, then run the allocation algorithm again with B = √xy.
+        Grid search over [√2/2 B, √2 B] for ckpt_opt over num_grids as in appendix A.
+        """
+        _, B_approx = run_chen_greedy(0)
+        B_min, B_max = math.floor(B_approx / math.sqrt(2)), math.ceil(B_approx * math.sqrt(2))
+        B_opt = math.inf
+        for B in range(B_min, B_max, (B_max - B_min) // num_grids):
+            ckpt, B_approx = run_chen_greedy(B)
+            if B_approx < B_opt:
+                B_opt = B_approx
```
Comment (FrankLeeeee) [inline colossalai/fx/passes/algorithms/ckpt_solver_chen.py:36]: do not start a variable with capital letter.

Comment (super-dainiu) [reply colossalai/fx/passes/algorithms/ckpt_solver_chen.py:36]: I was trying to follow the notations in the paper.

Comment (FrankLeeeee) [reply colossalai/fx/passes/algorithms/ckpt_solver_chen.py:36]: I understand, but it will  be good to stick with the python code standard.

Comment (super-dainiu) [reply colossalai/fx/passes/algorithms/ckpt_solver_chen.py:36]: I see

Comment (Cypher30) [reply colossalai/fx/passes/algorithms/ckpt_solver_chen.py:36]: Agree, just change the variables names

**Review by Cypher30 — APPROVED** [review]
Approve

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**